esp-hal/src/spi/master.rs

```diff
@@ -596,6 +596,14 @@ impl Config {
     fn raw_clock_reg_value(&self) -> Result<u32, ConfigError> {
         self.reg
     }
+
+    fn validate(&self) -> Result<(), ConfigError> {
+        // Max supported frequency is 80Mhz
+        if self.frequency > HertzU32::MHz(80) {
```

Comment:
Can't we not detect if the user requested a frequency, but the dividers are at their limits? SPI has a lower limit, too, something on the order of 80kHz that I hit in some tests.